Sources/PackageLoading/ToolsVersionLoader.swift

```diff
@@ -126,58 +297,287 @@ public class ToolsVersionLoader: ToolsVersionLoaderProtocol {
+        //                                                                         ☝️
+        // We know for sure that there is no numeric characters before `endIndexOfSpacingAfterCommentMarker`, so `specificationWithIgnoredTrailingContents[endIndexOfSpacingAfterCommentMarker...]` saves a bit of unnecessary searching.
+        
+        /// The label part of the Swift tools version specification with the whitespace sequence between the label and the version specifier.
+        /// - Note: For a misspelt Swift tools version specification `"// swift-too1s-version: 5.3.1"`, the label stops at the second `"o"`, so only `"swift-too"` is recognised as the label with no spacing following it.
+        let labelWithTrailingWhitespace = specificationWithIgnoredTrailingContents[endIndexOfSpacingAfterCommentMarker..<startIndexOfVersionSpecifier]
+        
+        /// The position of the first character in the spacing after the label part of the Swift version specification.
+        ///
+        /// Because there is no whitespace within the label, and because the spacing consists of only horizontal whitespace characters, so this position is the same as the position of the first whitespace character between the beginning of the label and the beginning of the version specifier. If no such whitespace character exists, then there is no spacing, and so this position is the `endIndex` of these sequence of characters (i.e. the starting position of the version specifier).
+        let startIndexOfSpacingAfterLabel = labelWithTrailingWhitespace.firstIndex(where: \.isWhitespace) ?? startIndexOfVersionSpecifier
+        
+        /// The label part of the Swift tools version specification.
+        /// - Note: For a misspelt Swift tools version specification `"// swift-too1s-version: 5.3.1"`, the label stops at the second `"o"`, so only `"swift-too"` is recognised as the label.
+        let label = labelWithTrailingWhitespace[..<startIndexOfSpacingAfterLabel]
+        
+        /// The spacing between the label part of the Swift tools version specification and the version specifier.
+        /// - Note: For a misspelt Swift tools version specification `"// swift-too1s-version: 5.3.1"`, the label stops at the second `"o"`, and the version specifier starts from the first `"1"`, so no spacing is recognised.
+        let spacingAfterLabel = labelWithTrailingWhitespace[startIndexOfSpacingAfterLabel...]
+        
+        /// The position of the version specifier's terminator.
+        ///
+        /// The terminator can be either a `";"` or a line terminator. If no such character exists, then this position is the `endIndex` of the Swift tools version specification.
+        let indexOfVersionSpecifierTerminator = specificationWithIgnoredTrailingContents[startIndexOfVersionSpecifier...].firstIndex(where: { $0 == ";" } ) ?? specificationWithIgnoredTrailingContents.endIndex
```

Comment:
Is the introduction of a comment separator here new in this PR?  I didn't see a facility for this in the earlier code.  If it's new, is it needed?  It seems to add complexity and I'm not sure whether it's important to add more comments using specialized syntax.

Reply:
The use of ";" isn't new. It's just now more visible in the logic instead of buried in the regex. The regex in the main branch (49cfc46bc5defd3ce8e0c0261e3e2cb475bcdb91, the commit this PR is based on currently) is [`^// swift-tools-version:(.*?)(?:;.*|$)`](https://github.com/WowbaggersLiquidLunch/swift-package-manager/blob/49cfc46bc5defd3ce8e0c0261e3e2cb475bcdb91/Sources/PackageLoading/ToolsVersionLoader.swift#L182), which looks for either a ";" or a line terminator after the version specifier. The `*?` pattern matches wildcard as few times as possible, so the first ";" after "swift-tools-version:" is considered as a terminator in the old/existing rule.

Reply:
Got it — I had missed that.  Yes, that's quite subtle indeed.  I really like how clear and well-commented it is now in your rewrite.